compiler/src/iree/compiler/Codegen/LLVMGPU/Passes.cpp

```diff
@@ -78,8 +78,10 @@ static void addBufferizePasses(OpPassManager &passManager) {
   passManager.addPass(createCSEPass());
 }
 
-static void tileAndDistributeToWorkgroup(OpPassManager &pm) {
+static void tileAndDistributeToWorkgroup(OpPassManager &pm,
+                                         bool specializeWorkgroup) {
   pm.addPass(createTileAndDistributeToWorkgroupsPass());
+  if (specializeWorkgroup) pm.addPass(createWorkgroupSpecializationPass());
```

Comment:
can this be called directly from the pipelines that need it rather than being in `tileAndDistributeToWorkgroup`. Would the clean up passes running after tile and distribute break things?

Reply:
I can try, but I didn't do it because the function does the bufferization too, which I didn't test the pass after it.

Reply:
> I can try, but I didn't do it because the function does the bufferization too, which I didn't test the pass after it.

It shouldn't be the case for the cases we care about

Reply:
Sounds good. Let me try.